Expose ability to set certificate renewal target times (#453)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/proc_configuring-tls-certificates-duration.adoc

[id="configuring-tls-certificates-duration_{context}"]
= Configuring TLS certificates duration

[role="_abstract"]

> You can configure the duration of the TLS certificates to use with {Project} ({ProjectShort}) to fit your deployment needs.
> For this purpose, modify the `ServiceTelemetry` object and configure the `certificates` parameter.

Reviewer: ```suggestion
To configure the duration of the TLS certificates to use with {Project} ({ProjectShort}), modify the `ServiceTelemetry` object and configure the `certificates` parameter.
```

.Prerequisites

> * Verify you don't have an instance of Service Telemetry Operator already deployed.

Reviewer: ```suggestion
* You didn't deploy an instance of Service Telemetry Operator already.
```

> NOTE:: Once created the `ServiceTelemetry` object, the required certificates and their secrets for {Project} ({ProjectShort}) are created.

Reviewer 2: ```suggestion
NOTE:: Once created the `ServiceTelemetry` object, the required certificates and their secrets for {ProjectShort} are created.
```

Reviewer 2: STF was defined in the intro, so you can use ProjectShort everywhere after in the same procedure.

Reviewer: When you create the `ServiceTelemetry` object, the required certificates and their secrets for {ProjectShort} are also created.

> In order to modify these you must follow a procedure similar to the one present in xref:assembly-renewing-the-amq-interconnect-certificate_assembly[]

Reviewer: ```suggestion
For more information about how to modify the certificates and the secrets, see: xref:assembly-renewing-the-amq-interconnect-certificate_assembly[]
```

> The following procedure is valid for new {Project} ({ProjectShort}) deployments.

Reviewer 2: ```suggestion
The following procedure is valid for new {ProjectShort} deployments.
```

.Procedure

> To override the duration of the TLS certificates, define which certificate you want to override.

Reviewer: ```suggestion
To edit the duration of the TLS certificates, you can set the Elasticsearch `endpointCertDuration`, for example `26280h` for 3 years, and set the QDR `caCertDuration`, for example `87600h` for 10 years.
```

For example, set the Elasticsearch endpoint certificate duration to 3 years by setting `endpointCertDuration` to `26280h`
and set QDR CA certificate duration to 10 years by setting `caCertDuration` to `87600h`.

> Keep the default value for the other certificates (CA certificate for Elasticsearch and endpoint certificate for QDR will keep the value of 8 years):

Reviewer: ```suggestion
You can use the default value of 8 years for the CA certificate for Elasticsearch and endpoint certificate:
```

+
[source,yaml,options="nowrap",role="white-space-pre"]
----
$ oc apply -f - <<EOF
apiVersion: infra.watch/v1beta1
kind: ServiceTelemetry
metadata:
  name: default
  namespace: service-telemetry
spec:
  backends:
    events:
      elasticsearch:
        enabled: true
        certificates:
          endpointCertDuration: 26280h
  transport:
    qdr:
      enabled: true
      certificates:
        caCertDuration: 87600h
EOF
----

.Verification

> . Verify that the desired expiry date appears in the certificates:

Reviewer: ```suggestion
. Verify that the expiry date for the certificates is correct:
```

[id="configuring-tls-certificates-duration_{context}"]
= Configuring TLS certificates duration

[role="_abstract"]
You can configure the duration of the TLS certificates to use with {Project} ({ProjectShort}) to fit your deployment needs.
For this purpose, modify the `ServiceTelemetry` object and configure the `certificates` parameter.

.Prerequisites

* Verify you don't have an instance of Service Telemetry Operator already deployed.

NOTE:: Once created the `ServiceTelemetry` object, the required certificates and their secrets for {Project} ({ProjectShort}) are created.

> In order to modify these you must follow a procedure similar to the one present in xref:xref:assembly-renewing-the-amq-interconnect-certificate_assembly[].

Reviewer 2: ```suggestion
In order to modify these you must follow a procedure similar to the one present in xref:assembly-renewing-the-amq-interconnect-certificate_assembly[]
```

> The following procedure is valid for greenfield {Project} ({ProjectShort}) deployments.

Reviewer 2: ```suggestion
The following procedure is valid for new {Project} ({ProjectShort}) deployments.
```

.Procedure

> +

Reviewer 2: ```suggestion
```

To override the duration of the TLS certificates, define which certificate you want to override.

> In this example, set Elasticsearch endpoint certificates duration to 3 years by setting `endpointCertDuration` to `26280h`

Reviewer 2: ```suggestion
For example, set the Elasticsearch endpoint certificate duration to 3 years by setting `endpointCertDuration` to `26280h`
```

> and set QDR CA Certificates duration to 10 years by setting `caCertDuration` to `87600h`.

Reviewer 2: ```suggestion
and set QDR CA certificate duration to 10 years by setting `caCertDuration` to `87600h`.
```

> Keep the default for the other certificates (CA certificates for Elasticsearch and endpoint certificates for QDR will keep the value of 8 years):

Reviewer 2: ```suggestion
Keep the default value for the other certificates (CA certificate for Elasticsearch and endpoint certificate for QDR):
```
